debug/tests/esm.html

```diff
@@ -6,7 +6,7 @@
   <body>
     <div id="map"></div>
     <script type="module">
-      import L from '../../dist/leaflet-src.esm.js';
+      import * as  L from '../../src/Leaflet.js';
```

Comment:
For now I'd like to leave this out of scope. Actually loading from source is problematic as we also have circular references. 
```suggestion
      import L from '../../dist/leaflet-src.esm.js';
```

Reply:
What's the problem with circular references? In my experience CR's are only an issue if circular access is needed synchronously at initialization time. Admittedly I did not port the entire test suite to confirm but when I tried this and a few other examples in the debug directory Leaflet seemed to work fine, and code that only works with a non-standard build tool is exactly the kind of thing I'm suggesting we take effort to avoid!

Reply:
I understand your enthusiasm to get this done, but I'd really like to give things a bit more time to bake. Moving Leaflet's distribution to ESM is already a big step forward. I also don't want to get rid the distribution just yet, as I am still on the fence over things like adding TypeScript support.

Reply:
fwiw I was not necessarily suggesting doing away with dist, I just want to be able to import leaflet natively for a better dev experience in my consuming applications, and to be able to bundle upstream as desired knowing leaflet's source is standard javascript (can you tell I vote no on typescript? heh).

Reply:
Ideally we don't adopt TypeScript. I'd prefer to adopt the [Type Annotations proposal](https://github.com/tc39/proposal-type-annotations), but that will likely take a couple of years before we can see it shipped. So pragmatism might be the reason to adopt TypeScript, at least for the short term.

Reply:
was not aware of that proposal, thanks for the link. i've just been bitten by things going out of style in js world enough times to have learned to favor standard / off-the-shelf stuff over the current hotness. ts has been around a while now but imho pragmatism in reusable modules means resisting the urge to go beyond the bleeding edge of what is standard at any given time. don't get me wrong, i love experimenting i just try to do it at the application level

Reply:
Yeah, it's just that a lot of people (including myself) really appreciate to have type-safety as it simply eliminates a whole class of bugs. Pretty sure if we were to adopt this for Leaflet we'd find a couple of golden ones :)

Reply:
i hear that for sure, types are an amazing developer tool, but there are tradeoffs right now that don't seem worth it to me: back to build tool lock in, massive dev dep trees, less accessible code, difficulty debugging at runtime, false sense of security etc. perhaps some of these are misconceptions or interdependent, just misc observations from past encounters with ts